paperqa/llms.py

```diff
@@ -34,36 +36,24 @@
     AsyncAnthropic = Any
     ContentBlockDeltaEvent = Any
 
-# only works for python 3.11
-# def guess_model_type(model_name: str) -> str:
-#     """Guess the model type from the model name for OpenAI models"""
-#     import openai
-
-#     model_type = get_type_hints(
-#         openai.types.chat.completion_create_params.CompletionCreateParamsBase
-#     )["model"]
-#     model_union = get_args(get_args(model_type)[1])
-#     model_arr = list(model_union)
-#     if model_name in model_arr:
-#         return "chat"
-#     return "completion"
-
-# def is_openai_model(model_name):
-#     import openai
-
-#     model_type = get_type_hints(
-#         openai.types.chat.completion_create_params.CompletionCreateParamsBase
-#     )["model"]
-#     model_union = get_args(get_args(model_type)[1])
-#     model_arr = list(model_union)
-
-#     complete_model_types = get_type_hints(
-#         openai.types.completion_create_params.CompletionCreateParamsBase
-#     )["model"]
-#     complete_model_union = get_args(get_args(complete_model_types)[1])
-#     complete_model_arr = list(complete_model_union)
-
-#     return model_name in model_arr or model_name in complete_model_arr
+litellm.vertex_ai_safety_settings = [
```

Comment:
We should not be specify this at the package level, given others install us. We need to let callers specify global behaviors like this.

What if someone installs `paper-qa`, but doesn't realize we removed safeties?

Let's move this to either a helper function or into our tests?

Reply:
Ofc for sure

Reply:
TBH - I think this is required for basic functionality. Their safety stuff trips up on all kinds of benign scientific words. Like a paper about breast cancer will give extremely opaque errors in paper-qa. I can't imagine a use-case where you would want to filter this strongly. Note that they apply other safety filtering (e.g., just turning this setting down won't let you make meth or bombs).

Reply:
makes sense @whitead  -- I compromised by making this a default setting that the user can override, but moved it out of module scope.

Reply:
I agree with both of you. People won't always use VertexAI, but when they are using it, it should be specified. I like Mike's idea of placing it to settings where it's obviously configured, instead of unexpectedly/"silently" at `import` time